src/praisonai/praisonai/gateway/server.py

```
@@ -288,6 +291,30 @@ async def start(self) -> None:
             logger.warning("Gateway already running")
             return
         
+        # Preflight port collision check
+        from .port_utils import check_port_available, GatewayPIDLock, format_collision_error
+        
+        pid_lock = GatewayPIDLock()
+        
+        # Check if port is available
+        port_available, pid_using_port = check_port_available(self._host, self._port)
```

Ruff (0.15.14)

[warning] Unpacked variable `pid_using_port` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)